### PR TITLE
TEAM TEACHING: use emplace_back and pre-reserved sizes for vectors

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26ef6af009ba0c616b54b62a0bc810317de4db9d available at: https://gist.github.com/ca62c51cf9ab515c01c4<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5079.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5079.dmg

File: libraries/render/src/render/DrawTask.cpp [hunk 1]
```diff
@@ -247,13 +247,14 @@ template <> void render::jobRun(const DrawOpaque& job, const SceneContextPointer
...
     for (auto id : items) {
-        inItems.push_back(ItemIDAndBounds(id));
+        inItems.emplace_back(ItemIDAndBounds(id));
     }
     ItemIDsBounds& renderedItems = inItems;
 
     renderContext->_numFeedOpaqueItems = renderedItems.size();
 
     ItemIDsBounds culledItems;
+    culledItems.reserve(inItems.size());
```
Comment (samcake) [inline libraries/render/src/render/DrawTask.cpp]: You could do that only in the if block on the size() of "renderedItems"

File: libraries/render/src/render/DrawTask.cpp [hunk 2]
```diff
@@ -265,6 +266,7 @@ template <> void render::jobRun(const DrawOpaque& job, const SceneContextPointer
 
 
     ItemIDsBounds sortedItems;
+    sortedItems.reserve(culledItems.size());
```
Comment (samcake) [inline libraries/render/src/render/DrawTask.cpp]: You could do that only in the if block on the size() of "renderedItems"

File: libraries/render/src/render/DrawTask.cpp [hunk 3]
```diff
@@ -313,13 +315,14 @@ template <> void render::jobRun(const DrawTransparent& job, const SceneContextPo
...
     for (auto id : items) {
-        inItems.push_back(id);
+        inItems.emplace_back(id);
     }
     ItemIDsBounds& renderedItems = inItems;
 
     renderContext->_numFeedTransparentItems = renderedItems.size();
 
     ItemIDsBounds culledItems;
+    culledItems.reserve(inItems.size());
```
Comment (samcake) [inline libraries/render/src/render/DrawTask.cpp]: You could do that only in the if block on the size() of "renderedItems"

**Pull request merged**